### PR TITLE
Making it work on MacOS

### PR DESCRIPTION
**Pull request by Pedro-Beirao**
As stated on the #1270 issue, I'm fixing MacOS support.

Changes:
1. The **lstdc++** library was deprecated several years ago (Source: https://developer.apple.com/forums/thread/113746)
2. **nil** has to be replaced by **lua_nil** (Source: https://github.com/mamedev/mame/commit/f191d038ec1fa110e06374ff79e345c91b3365d7)
3. **sol.hpp** needs to include *map* and *unordered_map* (Source: https://stackoverflow.com/questions/15733163/c-error-unordered-map-does-not-name-a-type) (It might also work if you only include *unordered_map*)

I also had a few problem with some *#include* statements with the **ftgl** library but thats something that someone trying to compile has to deal with imo.

You should try to compile this to Windows to check if there are no problems, before merging, obviously.

As stated on the issue I opened, Im available to maintain support for MacOS

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -206,7 +206,7 @@ target_link_libraries(slade
 	${LUA_LIBRARIES}
 	${MPG123_LIBRARIES}
 	${fmt_LIBRARIES}
-	-lstdc++fs
+
```
Comment (alexey-lysiuk) [inline src/CMakeLists.txt:209]: Doesn’t it break linking on Linux?

File: src/Scripting/Export/Archive.cpp [hunk 2]
```diff
@@ -39,6 +39,7 @@
 #include "Utility/StringUtils.h"
 #include "thirdparty/sol/sol.hpp"
 
+
```
Comment (alexey-lysiuk) [inline src/Scripting/Export/Archive.cpp:42]: Please remove this useless change.

**Review by alexey-lysiuk — CHANGES_REQUESTED** [review]
Please delete `.DS_Store` files.

File: src/CMakeLists.txt (hunk 1 above)
Comment (Pedro-Beirao) [reply src/CMakeLists.txt:209]: I have not tested on Linux, I was hopping someone would give me feedback about that.
Deleting it is necessary for it to work on MacOS.

File: src/Scripting/Export/Archive.cpp (hunk 2 above)
Comment (Pedro-Beirao) [reply src/Scripting/Export/Archive.cpp:42]: I was testing something and forgot that. I'll fix that and the .DS_STORE files

File: src/CMakeLists.txt (hunk 1 above)
Comment (Pedro-Beirao) [reply src/CMakeLists.txt:212]: Ok, now it check if its macOS or not

**Comment by sirjuddington** [comment]
Compiles fine in windows still, if @alexey-lysiuk doesn't have any other issues with it i'll merge it in

File: src/CMakeLists.txt (hunk 1 above)
Comment (alexey-lysiuk) [reply src/CMakeLists.txt:212]: I cannot check right now, but is it a correct CMake syntax?

**Comment by Pedro-Beirao** [comment]
@alexey-lysiuk you were right about the cmake file. I can't use an *if* statement inside of the *target_link_libraries*
I changed it and it should be fine

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -192,22 +192,40 @@ add_executable(slade WIN32 MACOSX_BUNDLE
...
-	${OPENGL_LIBRARIES}
-	${GLEW_LIBRARY}
-	${CURL_LIBRARIES}
-	${LUA_LIBRARIES}
-	${MPG123_LIBRARIES}
-	${fmt_LIBRARIES}
-	-lstdc++fs
-)
+if(APPLE) 
+	target_link_libraries(slade
```
Comment (alexey-lysiuk) [inline src/CMakeLists.txt:196]: `target_link_libraries()` can be used several times with the same target. Please restore the initial version without `stdc++fs`, and add it for Linux only with a separate call.

Actually, it should be added on any platform if we link with stdc++ standard library implementation.
However, proper detection of filesystem support in general requires a [separate module](https://github.com/vector-of-bool/CMakeCM/blob/6cb8a6ada09eb40c271d2f83a032e03cd4005fa3/modules/FindFilesystem.cmake).

Comment (Pedro-Beirao) [reply src/CMakeLists.txt]: Done

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -192,22 +192,25 @@ add_executable(slade WIN32 MACOSX_BUNDLE
...
+		${wxWidgets_LIBRARIES}
+		${FREEIMAGE_LIBRARIES}
+		${SFML_LIBRARY}
+		${FTGL_LIBRARIES}
+		${OPENGL_LIBRARIES}
+		${GLEW_LIBRARY}
+		${CURL_LIBRARIES}
+		${LUA_LIBRARIES}
+		${MPG123_LIBRARIES}
+		${fmt_LIBRARIES}
```
Comment (alexey-lysiuk) [inline src/CMakeLists.txt:209]: Please avoid useless changes in spacing like this altered indent and empty line. It’s much harder to review such changes. Git diff and blame features will produce unneeded noise.

**Comment by sirjuddington** [comment]
Closing this one since #1279 replaced it, but thank you @Pedro-Beirao for helping out with the macos stuff

**Pull request closed without merging**